src/language/parser.ts

```diff
@@ -1481,12 +1486,42 @@ export class Parser {
       this.expectToken(TokenKind.COLON);
       this.expectToken(TokenKind.PAREN_R);
     }
-    return this.node<SchemaCoordinateNode>(start, {
-      kind: Kind.SCHEMA_COORDINATE,
-      ofDirective,
+
+    if (ofDirective && !argumentName) {
+      return this.node<DirectiveCoordinateNode>(start, {
+        kind: Kind.DIRECTIVE_COORDINATE,
+        name,
+      });
+    }
+
+    if (ofDirective && argumentName) {
+      return this.node<DirectiveArgumentCoordinateNode>(start, {
+        kind: Kind.DIRECTIVE_ARGUMENT_COORDINATE,
+        name,
+        argumentName,
+      });
+    }
+
+    if (!ofDirective && memberName && argumentName) {
+      return this.node<ArgumentCoordinateNode>(start, {
+        kind: Kind.ARGUMENT_COORDINATE,
+        name,
+        memberName,
+        argumentName,
+      });
+    }
+
+    if (!ofDirective && memberName && !argumentName) {
+      return this.node<MemberCoordinateNode>(start, {
+        kind: Kind.MEMBER_COORDINATE,
+        name,
+        memberName,
+      });
+    }
+
+    return this.node<TypeCoordinateNode>(start, {
+      kind: Kind.TYPE_COORDINATE,
       name,
```

Comment:
this is a bit stinky, it seems like it would be more idiomatic to delegate parsing each intermediate node into its own function. [Y/n]?

Reply:
This LGTM but I'm really not the best TS person do don't take me into account too much ^^